src/cavendish_particle_tracks/_widget.py

```diff
@@ -111,6 +139,10 @@ def __init__(self, napari_viewer: napari.viewer.Viewer):
         self.mag_a = -1.0e6
         self.mag_b = -1.0e6
 
+        @self.viewer.layers.events.connect
+        def _on_layerlist_changed(event):
+            self.set_btn_availability()
+
```

Comment:
Indentation is because of an issue with using events in Napari plugins where the decorators won't work if used outside of __init__, as "self.viewer.layers" hasn't been defined at the moment the decorator would otherwise be used.

Depending on the event, you can get away without using a decorator - but when I was testing out the Napari events library, it seemed that the list of events available via the decorator, versus declaring the function outside of init and connecting at the bottom of __init__ with .connect were different. Might be worth verifying my claim though. It was a few weeks back and my memory of what exact combinations of events I tested is foggy.

This method does actually get called, even though Python/VSCode will tell you that it's an unused declaration. It's what prevents the user from using the program until they've both loaded an image (the "splash screen" and added a particle to the table.